docs/developer-docs/latest/development/backend-customization/controllers.md

```diff
@@ -277,6 +273,96 @@ async delete(ctx) {
 ::::
 :::::
 
+::::: details Collection type examples (Replacing a core action)
+
+:::: tabs card
+
+::: tab find()
+
+```js
+async find(ctx) {
+  const { query } = ctx;
+  // some logic here
+  const entity = await strapi.service('api::restaurant.restaurant').find(query);
+  const sanitizedEntity = await this.sanitizeOutput(entity, ctx);
+  // some logic here
+
+  return this.transformResponse(sanitizedEntity.results, {
+    pagination: sanitizedEntity.pagination,
+  });
+}
+```
+
+:::
+
+::: tab findOne()
+
+```js
+async findOne(ctx) {
+  const { id } = ctx.params;
+  const { query } = ctx;
+  // some logic here
+  const entity = await strapi.service('api::restaurant.restaurant').findOne(id, query);
+  const sanitizedEntity = await this.sanitizeOutput(entity, ctx);
+  // some logic here
+
+  return this.transformResponse(sanitizedEntity);
+}
+```
+
+:::
+
+::: tab create()
```

Comment:
This doesn't take into account if files are passed and won't take into account any query params. See the core one here: https://github.com/strapi/strapi/blob/09757686b0ddaec8609e2a5b657e6159a0a8fe5e/packages/core/strapi/lib/core-api/controller/collection-type.js#L45-L67